### PR TITLE
Update PyROS Solver and Webpage Documentation

### PR DESCRIPTION
**Pull request by shermanjasonaf**
## Summary/Motivation:

Updates to PyROS online documentation, and docstrings for the PyROS solver and `UncertaintySet` subclasses.

### Legal Acknowledgement

By contributing to this software project, I have read the [contribution guide](https://pyomo.readthedocs.io/en/stable/contribution_guide.html) and agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the BSD license.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

File: doc/OnlineDocs/contributed_packages/pyros.rst [hunk 1]
```diff
@@ -104,118 +132,161 @@ PyROS Solver Interface
...
+In this section, we illustrate the usage of PyROS with a modeling example.
+The deterministic problem of interest is called *hydro* 
+(available `here <https://www.gams.com/latest/gamslib_ml/libhtml/gamslib_hydro.html>`_),
+a QCQP taken from the
+`GAMS Model Library <https://www.gams.com/latest/gamslib_ml/libhtml/>`_.
+We have converted the model to Pyomo format using the
+`GAMS Convert tool <https://www.gams.com/latest/docs/S_CONVERT.html>`_.
+
+The *hydro* model features 31 variables,
+of which there are 13 are degrees of freedom and 18 state variables.
```
Comment (jsiirola) [inline doc/OnlineDocs/contributed_packages/pyros.rst:266]: "of which there are 13 are degrees of freedom and 18 state variables" -> "of which 13 are degrees of freedom and 18 are state variables"

File: doc/OnlineDocs/contributed_packages/pyros.rst [hunk 2]
```diff
@@ -322,33 +403,57 @@ The deterministic Pyomo model for *hydro* is shown below.
...
 
   >>> # === Construct the desirable uncertainty set ===
   >>> box_uncertainty_set = pyros.BoxSet(bounds=bounds)
 
 Step 3: Solve with PyROS
 ^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-PyROS requires the user to supply one local and one global NLP solver to be used for solving sub-problems. For convenience, we shall have PyROS invoke BARON as both the local and the global NLP solver.
+PyROS requires the user to supply one local and one global NLP solver to be
+used for solving sub-problems.
```
Comment (jsiirola) [inline doc/OnlineDocs/contributed_packages/pyros.rst:454]: "to be used" -> "to use"

File: doc/OnlineDocs/contributed_packages/pyros.rst [hunk 3]
```diff
@@ -358,60 +463,81 @@ PyROS requires the user to supply one local and one global NLP solver to be used
   >>> global_solver = pyo.SolverFactory('baron')
 
 .. note::
-    Additional solvers to be used as backup can be designated during the ``solve`` statement via the config options ``backup_local_solvers`` and ``backup_global_solvers`` presented above.
-
-The final step in solving a model with PyROS is to designate the remaining required inputs, namely ``first_stage_variables`` and ``second_stage_variables``. Below, we present two separate cases.
+    Additional NLP optimizers to be used in the event the primary
```
Comment (jsiirola) [inline doc/OnlineDocs/contributed_packages/pyros.rst:466]: "to be used" -> "can be automatically used"

File: doc/OnlineDocs/contributed_packages/pyros.rst [hunk 4]
```diff
@@ -358,60 +463,81 @@ PyROS requires the user to supply one local and one global NLP solver to be used
...
+    subordinate local and global optimizers passed
```
Comment (jsiirola) [inline doc/OnlineDocs/contributed_packages/pyros.rst:467]: "local and global optimizers" -> "local or global optimizer"
...
"do not successfully" -> "does not successfully"

File: doc/OnlineDocs/contributed_packages/pyros.rst [hunk 5]
```diff
@@ -358,60 +463,81 @@ PyROS requires the user to supply one local and one global NLP solver to be used
...
+    to the PyROS :meth:`~pyomo.contrib.pyros.PyROS.solve` method
+    do not successfully solve a subproblem to an appropriate termination
+    condition can be provided through the optional keyword arguments
+    ``backup_local_solvers`` and ``backup_global_solvers``.
```
Comment (jsiirola) [inline doc/OnlineDocs/contributed_packages/pyros.rst:471]: "condition can be provided through the optional keyword arguments ``backup_local_solvers`` and ``backup_global_solvers``." --> "condition.  These alternative solvers are provided through the optional keyword arguments ``backup_local_solvers`` and ``backup_global_solvers``."

**Review by jsiirola — COMMENTED** [review]
This looks good.  Just a few typos / grammatical edits.

**Comment by shermanjasonaf** [comment]
Ready for review.

**Review by mrmundt — APPROVED** [review]

**Review by jsiirola — APPROVED** [review]

**Pull request merged**